fenix/app/src/main/java/org/mozilla/fenix/experiments/ResearchSurfaceDialogFragment.kt

```diff
@@ -0,0 +1,114 @@
+ */
+
+class ResearchSurfaceDialogFragment : DialogFragment() {
+    private val args by navArgs<ResearchSurfaceDialogFragmentArgs>()
+    private lateinit var bundleArgs: Bundle
+
+    /**
+     * A callback to trigger the 'Take Survey' button of the dialog.
+     */
+    var onAccept: () -> Unit = {}
+
+    /**
+     * A callback to trigger the 'No Thanks' button of the dialog.
+     */
+    var onDismiss: () -> Unit = {}
+
+    @SuppressLint("SourceLockedOrientationActivity")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(STYLE_NO_TITLE, R.style.ResearchSurfaceDialogStyle)
+        activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+
+        bundleArgs = args.toBundle()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+
+        val messageText =
+            bundleArgs.getString(KEY_MESSAGE_TEXT) ?: getString(R.string.nimbus_survey_message_text)
+        val acceptButtonText = bundleArgs.getString(KEY_ACCEPT_BUTTON_TEXT)
+            ?: getString(R.string.preferences_take_survey)
+        val dismissButtonText = bundleArgs.getString(KEY_DISMISS_BUTTON_TEXT)
+            ?: getString(R.string.preferences_not_take_survey)
+
+        setContent {
+            FirefoxTheme {
+                ResearchSurfaceSurvey(
+                    messageText = messageText,
+                    onAcceptButtonText = acceptButtonText,
+                    onDismissButtonText = dismissButtonText,
+                    onDismiss = {
+                        onDismiss
+                    },
+                    onAccept = {
+                        onAccept
+                    },
+                )
+            }
+        }
+    }
+
+    companion object {
+        /**
+         * A builder method for creating a [ResearchSurfaceDialogFragment]
+         */
+        fun newInstance(
+            keyMessageText: String?,
+            keyAcceptButtonText: String?,
+            keyDismissButtonText: String?,
+        ): ResearchSurfaceDialogFragment {
+            val fragment = ResearchSurfaceDialogFragment()
+            val arguments = fragment.arguments ?: Bundle()
+
+            with(arguments) {
+                putString(KEY_MESSAGE_TEXT, keyMessageText)
+
+                putString(KEY_ACCEPT_BUTTON_TEXT, keyAcceptButtonText)
+
+                putString(KEY_DISMISS_BUTTON_TEXT, keyDismissButtonText)
+            }
+
+            fragment.arguments = arguments
+            fragment.isCancelable = false
+
+            return fragment
+        }
+
```

Comment:
To call `showNow` I think we also need a tag here defined.  Something like:
        const val FRAGMENT_TAG = "MOZAC_RESEARCH_SURFACE_DIALOG_FRAGMENT"